packages/salesforcedx-vscode-replay-debugger/src/index.ts

```diff
@@ -44,39 +45,27 @@ function registerDebugHandlers(): vscode.Disposable {
     async event => {
       if (event && event.session && event.session.type === DEBUGGER_TYPE) {
         if (event.event === GET_LINE_BREAKPOINT_INFO_EVENT) {
-          const sfdxApex = vscode.extensions.getExtension(
-            'salesforce.salesforcedx-vscode-apex'
-          );
-          if (sfdxApex && sfdxApex.exports) {
-            const lineBpInfo = await sfdxApex.exports.getLineBreakpointInfo();
-            event.session.customRequest(
-              LINE_BREAKPOINT_INFO_REQUEST,
-              lineBpInfo
-            );
-            console.log('Retrieved line breakpoint info from language server');
-          }
-        } else if (event.event === CHECKPOINT_INFO_EVENT) {
-          const eventBody = event.body as CheckpointMessage;
-          if (
-            eventBody &&
-            eventBody.sourceFile &&
-            eventBody.typeRef &&
-            eventBody.line
-          ) {
-            checkpointService.addCheckpointNode(
-              eventBody.sourceFile,
-              eventBody.typeRef,
-              eventBody.line
+          // Load the breakpoint line info stuff right here. Need to await this, otherwise
+          // we can (and will if they're stored) get breakpoint events before the line
+          // breakpoint information is fully loaded.
+          if (!BreakpointUtil.getInstance().hasLineNumberMapping()) {
```

Comment:
This looks like it’s here because retrieveLineBreakpointInfo could fail in activate, so you want to try again. Are you able to repro a scenario where this tries again and actually succeeds?

Reply:
Fixed.